src/dos/cdrom_image.cpp

```diff
@@ -557,78 +652,117 @@ bool CDROM_Interface_Image::LoadUnloadMedia(bool unload)
 	return true;
 }
 
-int CDROM_Interface_Image::GetTrack(int sector)
+track_iter CDROM_Interface_Image::GetTrack(const int sector)
 {
-	vector<Track>::iterator i = tracks.begin();
-	vector<Track>::iterator end = tracks.end() - 1;
-
-	while(i != end) {
-		Track &curr = *i;
-		Track &next = *(i + 1);
-		if (curr.start <= sector && sector < next.start) {
-			return curr.number;
+	// Setup our iterators
+	track_iter track(tracks.begin());
+	int lower_bound = track->start;
+	while(track != tracks.end()) {
+		const int upper_bound = track->start + track->length;
+		if (lower_bound <= sector && sector < upper_bound) {
+			break;
+		}
+		track++;
+		lower_bound = upper_bound;
+	}
+#ifdef DEBUG
+	// some games excessively query for track 1, so don't flood the console with these
+	if (track != tracks.begin() && track != tracks.end()) {
+		if (sector < track->start) {
+			LOG_MSG("CDROM: GetTrack at sector %d => in the pregap of track %d [pregap %d, start %d, end %d]",
+			        sector,
+			        track->number,
+			        prev(track)->start - prev(track)->length,
+			        track->start,
+			        track->start + track->length);
+		} else {
+			LOG_MSG("CDROM: GetTrack at sector %d => track %d [start %d, end %d]",
+			        sector,
+			        track->number,
+			        track->start,
+			        track->start + track->length);
 		}
-		i++;
+	} else if (track == tracks.end()) {
+		LOG_MSG("CDROM: GetTrack at sector %d => fell outside the bounds of our %d tracks",
+		        sector,
+				static_cast<int>(tracks.size()));
```

Comment:
Indeed! Fixes in cfd71f81

Reply:
The GCC build on Windows choked on `%zu`; new change in 7c4885c4 tries to use `#define __USE_MINGW_ANSI_STDIO 1` to let MinGW use it's own functions that comply with C11.
I've added that to `configure.ac` so all our code can benefit from `%zu` (when building with GCC under Windows).

Reply:
I didn't expect GCC to have a problem with it… I googled a bit and some people recommend using `_POSIX_C_SOURCE` instead of `__USE_MINGW_ANSI_STDIO` - I have no strong preference, but MINGW macro, being internal to the compiler, seems a bit scary.

GCC 9.2 on Fedora 30 defaults to `200809L`, which corresponds to IEEE Standard 1003.1-2008 (I have no idea which POSIX version added `%zu`).